pkg/volume/util/volumehelper/volumehelper.go

```diff
@@ -87,3 +90,56 @@ func GetUniqueVolumeNameFromSpec(
 			volumeName),
 		nil
 }
+
+// PostEventToPersistentVolumeClaim posts an event to the given PersistentVolumeClaim
+// API object with the given message and event type.
+func PostEventToPersistentVolumeClaim(
```

Comment:
We should use recorder.Eventf to post events because it will take care of the duplicate events too